use mysql reserved word  as a column name in hibernate with ShardingSphere-JDBC cause an exception

## Bug Report

### Which version of ShardingSphere did you use?
5.1.2 and 5.4.1

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-JDBC

### Expected behavior
get the query result

### Actual behavior
get the exception: You have an error in your SQL syntax

### Reason analyze (If you can)
when use reserved words in sql, whether use it directly or after dot ,shardingsphere-jdbc sql parser throw an exception,but the jdbc can support after dot 

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
select ruleitem0_.condition as conditio2_21_ from rule_item ruleitem0_ where ruleitem0_.type=?
throws an exception:
Caused by: org.apache.shardingsphere.sql.parser.exception.SQLParsingException: You have an error in your SQL syntax
	
this is not consistent with the result when using jdbc instead
also this query can be  execute on mysql server directly
### Example codes for reproduce this issue (such as a github link).

--- a/parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/BaseRule.g4
+++ b/parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/BaseRule.g4
@@ -126,7 +126,7 @@ identifier
     | UNDERSCORE_CHARSET
     | BQUOTA_STRING
     ;
-    
+
 identifierKeywordsUnambiguous
     : ACTION
     | ACCOUNT
@@ -522,6 +522,8 @@ identifierKeywordsUnambiguous
     | XML
     | YEAR
     | YEAR_MONTH
+    | CONDITION
+    | DESCRIBE
     ;
     
 identifierKeywordsAmbiguous1RolesAndLabels

--- a/test/it/parser/src/main/resources/case/dml/select.xml
+++ b/test/it/parser/src/main/resources/case/dml/select.xml
@@ -9478,4 +9478,24 @@
             </simple-table>
         </from>
     </select>
+
+    <select sql-case-id="select_with_reserved_word_with_table_ref">
+        <projections start-index="7" stop-index="19">
+            <column-projection name="condition" start-index="7" stop-index="19" >
+                <owner name="xxx" start-index="7" stop-index="9" />
+            </column-projection>
+        </projections>
+        <from start-index="26" stop-index="28">
+            <simple-table name="xxx" start-index="26" stop-index="28"/>
+        </from>
+    </select>
+
+    <select sql-case-id="select_with_reserved_word">
+        <projections start-index="7" stop-index="14">
+            <column-projection name="describe" start-index="7" stop-index="14" />
+        </projections>
+        <from start-index="21" stop-index="23">
+            <simple-table name="xxx" start-index="21" stop-index="23"/>
+        </from>
+    </select>
 </sql-parser-test-cases>

--- a/test/it/parser/src/main/resources/sql/supported/dml/select.xml
+++ b/test/it/parser/src/main/resources/sql/supported/dml/select.xml
@@ -280,4 +280,6 @@
     <sql-case id="select_with_index_hints1" value="SELECT * FROM t1 USE INDEX (i1) IGNORE INDEX (i2) USE INDEX (i2);" db-types="MySQL,Doris"/>
     <sql-case id="select_with_index_hints2" value="SELECT * FROM t1 USE INDEX () IGNORE INDEX (i2) USE INDEX (i1) USE INDEX (i2);" db-types="MySQL,Doris"/>
     <sql-case id="select_with_index_hints3" value="SELECT * FROM t1 USE INDEX (i1,i2) IGNORE INDEX (i2);" db-types="MySQL,Doris"/>
+    <sql-case id="select_with_reserved_word_with_table_ref" value="select xxx.condition from xxx" db-types="MySQL"/>
+    <sql-case id="select_with_reserved_word" value="select describe from xxx" db-types="MySQL"/>
 </sql-cases>

